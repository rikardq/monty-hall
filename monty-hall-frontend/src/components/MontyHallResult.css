.MontyHallResult {
    flex: 1;
    flex-direction: column;
    display: flex;
    background-color: #222;
    color: white;
    min-height: 100vh;
}