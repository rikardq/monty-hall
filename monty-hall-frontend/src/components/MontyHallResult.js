import React, {Component} from "react";
import './MontyHallResult.css';

export default class MontyHallResult extends Component {

    renderResult = (result, index) => {
        return (
            <tr key={index}>
                <td>{index + 1}</td>
                <td>{result.numberOfLosses}</td>
                <td>{result.numberOfWins}</td>
                <td>{result.gameStrategy}</td>
                <td>{result.numberOfSimulations}</td>
                <td>{result.winPercentage}</td>
            </tr>
        );
    };

    render() {
        const results = this.props.results;
        return (
            <div className="MontyHallResult">
                <h1>Results</h1>
                <table>
                    <tr>
                        <th>Run</th>
                        <th>Losses</th>
                        <th>Wins</th>
                        <th>Strategy</th>
                        <th>simulations</th>
                        <th>Win percentage</th>
                    </tr>
                    {results.map((result, index) => this.renderResult(result, index))}
                </table>

            </div>
        );
    }
}