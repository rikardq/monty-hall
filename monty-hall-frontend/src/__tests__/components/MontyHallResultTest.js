import React from 'react';
import MontyHallResult from "../../components/MontyHallResult";
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

it('renders without crashing', () => {
    const resultData = {
        numberOfLosses: 51,
        numberOfWins: 49,
        gameStrategy: 'RANDOM',
        numberOfSimulations: 100,
        winPercentage: 49.0
    };

    const component = shallow(<MontyHallResult results={ [resultData] }/>);
    expect(component).toMatchSnapshot();
});
