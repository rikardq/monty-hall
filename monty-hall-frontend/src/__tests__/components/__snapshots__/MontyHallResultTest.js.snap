// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="MontyHallResult"
>
  <h1>
    Results
  </h1>
  <table>
    <tr>
      <th>
        Run
      </th>
      <th>
        Losses
      </th>
      <th>
        Wins
      </th>
      <th>
        Strategy
      </th>
      <th>
        simulations
      </th>
      <th>
        Win percentage
      </th>
    </tr>
    <tr
      key="0"
    >
      <td>
        1
      </td>
      <td>
        51
      </td>
      <td>
        49
      </td>
      <td>
        RANDOM
      </td>
      <td>
        100
      </td>
      <td>
        49
      </td>
    </tr>
  </table>
</div>
`;
